### PR TITLE
CLOUDP-110780: update cleaner

### PR DESCRIPTION
**Pull request by leo-ri**
### All Submissions:

* [+] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: .github/actions/cleanup/entrypoint.sh [hunk 1]
```diff
@@ -13,6 +13,50 @@ delete_project() {
...
+    for elkey in $(echo "$list" | jq -c '.[]'); do
+        echo $elkey
+        pe_service_id=$(echo "$elkey" | jq -r '.id' )
+        for pe in $(echo "$elkey" | jq -r $endpoint_field' | .[]'); do
+            echo "private endpoint: $pe from $projectID will be deleted"
+
+            pe=$(echo "$pe" | sed 's/\//%2F/g')
+            delete_pe "$projectID" "$provider" "$pe_service_id" "$pe"
+        done
+        sleep 5 # fast operation we can wait
```
Comment (leo-ri) [inline .github/actions/cleanup/entrypoint.sh:47]: probably need to add real awaits

File: .github/workflows/cleanup-all.yml [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 # local run sample:
-# echo '{"ref":"HEAD/ref/CLOUDP-smth", "ref_type":"branch"}' > payload.json | act -j cleanup-all -e payload.json
-# ^ command will clean branch organization with name `CLOUDP-smth`
+# echo '{"ref":"HEAD/ref/CLOUDP-smth", "ref_type":"branch", "action":"workflow_dispatch", "inputs": {"clean":"true"} }' > payload.json | act -j cleanup-all -e payload.json
+
 on:
   workflow_dispatch:
+    inputs:
+      clean:
+        description: "Clean all(true) or old empty projects(false)?"
```
Comment (leo-ri) [inline .github/workflows/cleanup-all.yml:9]: I'm not sure how to explain it better in one line, this is ok? 
note: I believe for now better to live both cases before updating the ginkgo version and make sure after each is working. And after we can delete choses

**Review by igor-karpukhin — APPROVED** [review]
Looks good

File: .github/workflows/cleanup-all.yml (hunk 2 above)
Comment (chatton) [reply .github/workflows/cleanup-all.yml:9]: I think it's okay to leave it doing both still! I think it's clear enough what true & false do in this message.

**Review by chatton — APPROVED** [review]
LGTM! I think using MCLI makes things simpler here!

File: .github/actions/cleanup/Dockerfile [hunk 3]
```diff
@@ -8,6 +8,8 @@ RUN apk update && \
     apk add bash && \
     apk add --update coreutils && \
     rm -rf /var/cache/apk/*
+RUN curl -O -L https://github.com/mongodb/mongocli/releases/download/v1.21.0/mongocli_1.21.0_linux_x86_64.tar.gz && \
```
Comment (gssbzn) [inline .github/actions/cleanup/Dockerfile:11]: ❤️

**Review by fabritsius — APPROVED** [review]
Great job 👏

**Review by igor-karpukhin — APPROVED** [review]
Looks good!

**Pull request merged**